tests/rust/tcp-test/close/async_close.rs

```diff
@@ -0,0 +1,218 @@
+pub const AF_INET: i32 = libc::AF_INET;
+
+#[cfg(target_os = "linux")]
+pub const SOCK_STREAM: i32 = libc::SOCK_STREAM;
+
+//======================================================================================================================
+// Standalone Functions
+//======================================================================================================================
+
+/// Drives integration tests for async_close() on TCP sockets.
+pub fn run(libos: &mut LibOS, addr: &SocketAddrV4) -> Result<()> {
+    async_close_invalid_queue_descriptor(libos)?;
+    async_close_socket_twice(libos)?;
+    async_close_unbound_socket(libos)?;
+    async_close_bound_socket(libos, addr)?;
+    async_close_listening_socket(libos, addr)?;
+    async_close_accepting_socket(libos, addr)?;
+    async_close_connecting_socket(libos, addr)?;
+
+    Ok(())
+}
+
+/// Attempts to close an invalid queue descriptor.
+fn async_close_invalid_queue_descriptor(libos: &mut LibOS) -> Result<()> {
+    println!("{}", stringify!(async_close_invalid_queue_descriptor));
+
+    // Fail to close socket.
+    let e: Fail = libos
+        .async_close(QDesc::from(0))
+        .expect_err("close() an invalid socket should fail");
+
+    // Sanity check error code.
+    assert_eq!(e.errno, libc::EBADF, "async_close() failed with {}", e.cause);
+
+    Ok(())
+}
+
+/// Attempts to close a TCP socket multiple times.
+fn async_close_socket_twice(libos: &mut LibOS) -> Result<()> {
+    println!("{}", stringify!(async_close_socket_twice));
+
+    // Create an unbound socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+
+    // Succeed to close socket.
+    let qt: QToken = libos.async_close(sockqd)?;
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE => {},
+        Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+        Err(_) => anyhow::bail!("wait() should succeed with async_close()"),
+    }
+
+    // Fail to close socket.
+    let e: Fail = libos
+        .async_close(sockqd)
+        .expect_err("async_close() a socket twice should fail");
+
+    // Sanity check error code.
+    assert_eq!(e.errno, libc::EBADF, "async_close() failed with {}", e.cause);
+
+    Ok(())
+}
+
+/// Attempts to close a TCP socket that is not bound.
+fn async_close_unbound_socket(libos: &mut LibOS) -> Result<()> {
+    println!("{}", stringify!(async_close_unbound_socket));
+
+    // Create an unbound socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+
+    // Succeed to close socket.
+    let qt: QToken = libos.async_close(sockqd)?;
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Ok(qresult) if qresult.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE => {},
```

Comment:
Could you rename `qresult` to `qr` so that it follows the convetions on the source tree?